### DISCUSSION TITLE
Strange reverse behavior

### DISCUSSION THREAD

**Discussion by cschwenk99** [nebhead/PiFire#188] [Q&A] [answered]
Hoping someone can help me out here. I am fairly new to the Pi world and programming. I have some experience with DIY circuits and soldering and have a mechanical engineering background. I used Weber's newest PCB with PWM fan build and it was relatively uneventful until now. Had some expected troubleshooting right away, but then everything seemed to be working as it should. Now I am having issues trying to control the grill remotely. Running in Monitor mode seems to work just fine, grill goes to about 300F and holds. When I switch to "Startup", "Smoke" or "Hold" on my pc, phone browser or android app, the grill immediately shuts off. If I use the "Manual Mode" on the app, all the toggles do the opposite of what is expected (e.g. if i toggle the fan off, it turns on, or toggle auger on it turns off, etc). Is this evidence that my relay active levels are backwards? If not, any other ideas I could try? Right now I am running without LCD or inputs, they are on the way from China. Thanks in advance!

**Comment by nebhead** [comment, marked as answer]
When you go through the configuration wizard, did you select active high or active low triggers?   I believe that the PWM board is active high.

**Reply by cschwenk99** [reply]
Wow! Impressed with the response time! I had set them as active low. I just reloaded with active high and it works now! Thanks! Just a suggestion, I went back and looked at the hardware notes, and there are several notes about the special button board being active high, which I took to mean the pullups/pulldowns menu which is also labeled as low/high, but nothing I could find explicitly says the pwm pcb should be active high. I could certainly be missing something. Regardless, I am very excited about this project. Really a game changer!

**Reply by nebhead** [reply]
No problem!  Glad you are having fun with it.  Thanks for the feedback - I'll look into changing this in future revisions.
